src/app/public/modules/sky-tree-view/sky-tree-view.component.ts

```diff
@@ -0,0 +1,52 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ContentChild,
+  Input,
+  ViewEncapsulation
+} from '@angular/core';
+
+import {
+  TreeComponent
+} from 'angular-tree-component';
+
+import {
+  SkyTreeViewOptions
+} from './types/sky-tree-view-options';
+
+@Component({
+  selector: 'sky-tree-view',
+  templateUrl: './sky-tree-view.component.html',
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class SkyTreeViewComponent {
+
+  @Input()
+  public set options(value: SkyTreeViewOptions) {
+    this._options = value;
+  }
+
+  public get options(): SkyTreeViewOptions {
+    const defaultOptions: SkyTreeViewOptions = {
+      leafNodeSelectionOnly: false
```

Comment:
Do we have any other options planned in the near future? If not, it might be easier to provide `leafNodeSelectionOnly` as an input, so that change detection on that setting is easier to manage. This would also make it less confusing as to what the `options` are affecting (e.g., Are these options for the `tree-root` component, too?).

Reply:
No, these options do not affect the `tree-root` component, because these only exist to fill in gaps that the original tree component didn't meet. These particular options (`SkyTreeViewOptions`) are only used for extra functionality that our design docs specify. I'm worried that limiting it to just an input would keep us from growing that out in the future as our requirements change. On the other hand, if we changed this to just a a `leafNodeSelectionOnly` property, then we could simply add more input properties if requirements grow.
Do you have strong feelings on which way is better?

Reply:
I personally prefer single-value inputs because change detection is designed to work with immutable values. As you know, changing the property value of a complex object doesn't trigger change detection and causes confusion to consumers that aren't aware of this behavior.

My vote is to lean toward single-value inputs for components and leave complex configuration objects to public service methods.